os/kernel/log_dump/log_dump.c

```diff
@@ -239,3 +412,36 @@ size_t log_dump_read(FAR char *buffer, size_t buflen)
 
 	return ret;
 }
+
+int log_dump(int argc, char *argv[]) {
+	struct mq_attr attr;
+
+	attr.mq_maxmsg = 32;
+	attr.mq_msgsize = sizeof(struct log_dump_msg_s);
+	attr.mq_flags = 0;
+
+	/* Create binary manager message queue */
+	g_logdump_mq_fd = mq_open("logdump_s", O_RDWR | O_CREAT, 0666, &attr);
+	if (g_logdump_mq_fd == (mqd_t)ERROR) {
+		lldbg("Fail to open mq, errno %d. EXIT!\n", errno);
+		return 0;
+	}
+
+	log_dump_init();
+
+	struct log_dump_msg_s msg;
+	int nbytes;
+
+	while (true) {
+		nbytes = mq_receive(g_logdump_mq_fd, (char *)&msg, sizeof(struct log_dump_msg_s), NULL);
```

Comment:
mq_timedreceive & retry count is better option, in my opinion.
It should not affect to other process.

Reply:
As we wait till we get the message, so I hope timeout is not needed?